crates/miden-proving-service-client/Cargo.toml

```diff
@@ -17,7 +17,8 @@ crate-type = ["lib"]
 default = ["std"]
 std = ["miden-objects/std", "miden-tx/std"]
 tx-prover = ["miden-tx/async"]
-batch-prover = ["miden-tx/async", "dep:tokio"]
+batch-prover = ["miden-tx/async", "dep:tokio", "dep:miden-tx-batch-prover"]
+block-prover = ["miden-tx/async", "dep:tokio"]
```

Comment:
Why do `batch-prover` and `block-prover` need `miden-tx/async` dependency?

Also, why doesn't `tx-prover` need `dep:tokio` dependency.

Reply:
Those do not need it, I just removed it (only tx-prover use it, for the async version of the `TransactionProver` trait).

Regarding `tx-prover` and `tokio`. For the batch and blocks provers, we use tokio to wrap the client in a mutex (because of the specific use case on the node). For the tx-prover, i just use miden-tx's `RwLock`, though it might make sense to use just a mutex since we don't have an use case when we just need to read the value.

Reply:
> Regarding `tx-prover` and `tokio`. For the batch and blocks provers, we use tokio to wrap the client in a mutex (because of the specific use case on the node). For the tx-prover, i just use miden-tx's `RwLock`, though it might make sense to use just a mutex since we don't have an use case when we just need to read the value.

Makes sense. I think if we can apply a consistent methodology, that would be better - but not sure if it causes some kind of issues.

Reply:
I changed the tx-prover client to use Arc<Mutex<Client>>.